vm/vm.go

```diff
@@ -753,16 +749,16 @@ func ExecFuncBalancedBaseTokens() ExecFunc {
 		}
 
 		if in != out {
-			return ierrors.Wrapf(iotago.ErrInputOutputBaseTokenMismatch, "in %d, out %d", in, out)
+			return ierrors.WithMessagef(iotago.ErrInputOutputBaseTokenMismatch, "in %d, out %d", in, out)
 		}
 
 		for addr, returnSum := range inputSumReturnAmountPerAddress {
 			outSum, has := outputSimpleTransfersPerAddr[addr]
 			if !has {
-				return ierrors.Wrapf(iotago.ErrReturnAmountNotFulFilled, "return amount of %d not fulfilled as there is no output for %s", returnSum, addr)
+				return ierrors.WithMessagef(iotago.ErrReturnAmountNotFulFilled, "return amount of %d not fulfilled as there is no output for address (serialized) %s", returnSum, hexutil.EncodeHex([]byte(addr)))
```

Comment:
Why not use `addr.Bech32(vmParams.API.ProtocolParameters().Bech32HRP())`

Reply:
`addr` is of type `string` because its the result of calling `Key()` on the return address. We could reconstruct the address but that seemed like a bit too much so I thought why not just print the serialized address (which is equivalent to `Key()`).

Reply:
ah ok, then fine :+1: